boxoffice/models/invoice.py

```diff
@@ -72,6 +73,18 @@ def __init__(self, *args, **kwargs):
         self.invoice_no = gen_invoice_no(organization, country_code, self.invoiced_at)
         super(Invoice, self).__init__(*args, **kwargs)
 
+    @property
+    def is_final(self):
+        return self.status == INVOICE_STATUS.FINAL
+
+@event.listens_for(Invoice, 'before_update')
+def validate_immutable_finalized_invoice(mapper, connection, target):
+    """
+    Raises ValueError if a finalized invoice is being modified. Allows for the status to be toggled.
+    """
+    if target.is_final and get_history(target, 'status').deleted.count(INVOICE_STATUS.DRAFT) == 0:
```

Comment:
What does `get_history` do? Can you document what's happening here with a comment?